src/hardware/mixer.cpp

```diff
@@ -246,171 +235,65 @@ static void set_global_crossfeed(mixer_channel_t channel)
-	}
+	if (!mixer.reverb.enabled)
+		channel->SetReverbLevel(0.0f);
+	else if (channel->HasFeature(ChannelFeature::Synthesizer))
+		channel->SetReverbLevel(mixer.reverb.synthesizer_send_level);
+	else if (channel->HasFeature(ChannelFeature::DigitalAudio))
+		channel->SetReverbLevel(mixer.reverb.digital_audio_send_level);
 }
 
-static void configure_reverb()
+static void configure_reverb(std::string reverb_pref)
 {
-	const auto sect = static_cast<Section_prop *>(control->GetSection("mixer"));
-	assert(sect);
+	// "on" is an alias for the medium room size preset
+	if (reverb_pref == "on")
+		reverb_pref = "medium";
 
-	const std::string reverb_pref = sect->Get_string("reverb");
+	enum class Preset { off, tiny, small, medium, large, huge };
 
-	auto enable_reverb = [&reverb_pref](const ReverbPreset preset) {
-		mixer.reverb.mverb.setSampleRate(static_cast<float>(mixer.sample_rate));
-		set_reverb_preset(preset);
-		LOG_MSG("MIXER: Reverb enabled ('%s' preset)", reverb_pref.c_str());
-	};
-
-	if (reverb_pref == "off") {
-		mixer.reverb.enabled = false;
-
-	} else if (reverb_pref == "on") {
-		enable_reverb(ReverbPreset::Medium);
-
-	} else if (reverb_pref == "tiny") {
-		enable_reverb(ReverbPreset::Tiny);
+	const auto preset = enum_cast<Preset>(reverb_pref).value();
 
-	} else if (reverb_pref == "small") {
-		enable_reverb(ReverbPreset::Small);
+	auto &r = mixer.reverb; // short-hand reference
 
-	} else if (reverb_pref == "medium") {
-		enable_reverb(ReverbPreset::Medium);
+	const auto rate_hz = mixer.sample_rate.load();
 
-	} else if (reverb_pref == "large") {
-		enable_reverb(ReverbPreset::Large);
+	// Pre-computed (negative) decibel scalars
+	constexpr auto __5_2dB = 0.87f;
+	constexpr auto __6_0dB = 0.85f;
+	constexpr auto _12_0dB = 0.70f;
+	constexpr auto _18_4dB = 0.54f;
+	constexpr auto _24_0dB = 0.40f;
+	constexpr auto _36_8dB = 0.08f;
+	constexpr auto _37_2dB = 0.07f;
+	constexpr auto _38_0dB = 0.05f;
 
-	} else if (reverb_pref == "huge") {
-		enable_reverb(ReverbPreset::Huge);
-
-	} else {
-		LOG_WARNING("MIXER: Invalid reverb setting '%s', using off",
-		            reverb_pref.c_str());
-		mixer.reverb.enabled = false;
+	// clang-format off
+	switch (preset) { //         PDELAY  EARLY   SIZE DNSITY BWFREQ  DECAY DAMPLV   -SYNLV   -DIGLV HIPASSHZ RATE_HZ
+	case Preset::tiny:   r.Config(0.00f, 1.00f, 0.05f, 0.50f, 0.50f, 0.00f, 1.00f, __5_2dB, __5_2dB, 200.0f, rate_hz); break;
+	case Preset::small:  r.Config(0.00f, 1.00f, 0.17f, 0.42f, 0.50f, 0.50f, 0.70f, _24_0dB, _36_8dB, 200.0f, rate_hz); break;
+	case Preset::medium: r.Config(0.00f, 0.75f, 0.50f, 0.50f, 0.95f, 0.42f, 0.21f, _18_4dB, _37_2dB, 170.0f, rate_hz); break;
+	case Preset::large:  r.Config(0.00f, 0.75f, 0.75f, 0.50f, 0.95f, 0.52f, 0.21f, _12_0dB, _38_0dB, 140.0f, rate_hz); break;
+	case Preset::huge:   r.Config(0.00f, 0.75f, 0.75f, 0.50f, 0.95f, 0.52f, 0.21f, __6_0dB, _38_0dB, 140.0f, rate_hz); break;
+	case Preset::off:    r.enabled = false; break;
```

Comment:
This looks really neat now, @kcgen 🎉

Reply:
All your find tuning on display :chart_with_upwards_trend: !